WordPress/gutenberg
# Post a summary of the flaky tests to the commit

## What?

This is an attempt to bring more visibility to flaky tests during reviews. It will post a comment on the commit which has flaky tests.

## Why?

The original flaky tests reporter started with the idea of retrying failed tests to unblock contributors in PRs with flaky tests. However, a side-effect is that flaky tests tend to be overlooked and ignored by the author. This PR tries to fix that by posting the summary of flaky tests to the commit without blocking the PR authors.

## How?


Tons of improvements are in this PR:

- End-to-end testing workflows are merged into one. It seems like the only way to ensure only one run of the `report-flaky-tests` job will be run after all e2e jobs are finished.
- We'll bail out of the `report-flaky-tests` job if there isn't any flaky tests report.
- We no longer split artifacts into different names from different matrixes. It seems like GitHub allows uploading different files to the same artifact.

## Testing Instructions




I'm not aware of an easy way to test this. It's unfortunately a problem with custom github actions. The best way I can think of is to fork the repo and test it there.

See below for detailed instructions on how to test it in your own fork: https://github.com/WordPress/gutenberg/pull/45798#issuecomment-1319601801.

## Screenshots or screencast 
Here's what it might look like in the commit:

![image](https://user-images.githubusercontent.com/7753001/202118516-c77cf305-ca95-42ee-ba96-cbf6cb17d95c.png)

Or you can view it here: https://github.com/kevin940726/gutenberg/commit/b87c01c7e68ce50b1102d5754031651b19d2aaea#commitcomment-90061333
